### ISSUE TITLE
legend position export

### ISSUE THREAD

**Issue by falexwolf** [scverse/scanpy#335]
The previous `sc.pl.umap` etc. had an option to export legend positions via 'on data export'. We need a solution in the docs... Presumably just by exposing the cached positions to the user.

**Comment by fidelram** [comment]
With export legend positions, you mean when  `legend_loc='on data` ? It this is needed, the positions can be easily returned or saved. How this is useful?

**Comment by falexwolf** [comment]
Yes, a few people manually overlay different images and need the label positions for that. But I agree, it doesn't seem like a canonical case and we don't want the previous hackish solution for that. Maybe one could add the label positions in an unstructured annotation, though: `.uns['umap']['pos_louvain`]`. Wouldn't have a actual disadvantages, I guess.
